File: server/src/main/com/broll/gainea/server/core/utils/ProcessingUtils.java
package com.broll.gainea.server.core.utils;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public final class ProcessingUtils {
    private final static Logger Log = LoggerFactory.getLogger(ProcessingUtils.class);

    private final static int DEFAULT_PAUSE = 1000;

    public static int MAX_PAUSE = 0;

    public static void pause() {
        pause(DEFAULT_PAUSE);
    }

    public static void pause(int ms) {
        if (MAX_PAUSE > 0) {
            ms = Math.min(MAX_PAUSE, ms);
        }
        if (ms > 0) {
            try {
                Thread.sleep(ms);
            } catch (InterruptedException e) {
                throw new RuntimeException("shutdown");
            }
        }
    }

}

File: server/src/main/com/broll/gainea/server/core/utils/ProcessingUtils.kt
package com.broll.gainea.server.core.utils

import org.slf4j.LoggerFactory

object ProcessingUtils {
    private val Log = LoggerFactory.getLogger(ProcessingUtils::class.java)
    private const val DEFAULT_PAUSE = 1000
    var MAX_PAUSE = 0
    @JvmOverloads
    fun pause(ms: Int = DEFAULT_PAUSE) {
        var ms = ms
        if (MAX_PAUSE > 0) {
            ms = Math.min(MAX_PAUSE, ms)
        }
        if (ms > 0) {
            try {
                Thread.sleep(ms.toLong())
            } catch (e: InterruptedException) {
                throw RuntimeException("shutdown")
            }
        }
    }
}
